ssm_jax/kmeans.py

```diff
@@ -0,0 +1,147 @@
+from functools import partial
+
+import jax.numpy as jnp
+import jax.random as jr
+from jax import jit
+from jax import lax
+from jax import vmap
+
+
+def find_closest_cluster(X, centroids):
+
+    def assign(x):
+        distances = vmap(jnp.linalg.norm)(centroids - x)
+        min_distance = jnp.min(distances)
+        label = jnp.argmin(distances)
```

Comment:
this is searching twice. you could save a bit by doing 
```
label = jnp.argmin(distances)
min_distance = distances[label]
```

Reply:
Done